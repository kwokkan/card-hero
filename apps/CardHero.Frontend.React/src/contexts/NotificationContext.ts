import React from "react";
import { INotificationItem } from "../types/INotificationItem";

export interface INotificationContextProps {
    notifications: INotificationItem[];
    addNotification: (notification: INotificationItem) => void;
    removeNotification: (notification: INotificationItem) => void;
}

export const NotificationContext = React.createContext<INotificationContextProps>({
    notifications: [],
    addNotification: (_notification: INotificationItem) => { },
    removeNotification: (_notification: INotificationItem) => { }
});
