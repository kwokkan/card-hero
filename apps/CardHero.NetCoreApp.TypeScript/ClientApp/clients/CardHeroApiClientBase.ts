// no export as this is used by NSwag generated api clients
class CardHeroApiClientBase {
    getBaseUrl(defaultBaseUrl: string, baseUrl: string): string {
        if (!baseUrl) {
            return defaultBaseUrl;
        }

        if (baseUrl.endsWith('/')) {
            // remove trailing slash as NSwag uses '/' as part of the api url
            return baseUrl.substring(0, baseUrl.length - 2);
        }

        return baseUrl;
    }
}

module.exports = CardHeroApiClientBase;