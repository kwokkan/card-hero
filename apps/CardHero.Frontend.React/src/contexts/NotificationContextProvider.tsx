import React, { useContext, useState } from "react";
import { INotificationItem } from "../types/INotificationItem";
import { NotificationType } from "../types/NotificationType";
import { INotificationContextProps, NotificationContext } from "./NotificationContext";

interface INotificationContextProviderState {
    notifications: INotificationItem[];
}

function NotificationContextProvider(props: any) {
    const [state, setState] = useState<INotificationContextProviderState>({
        notifications: [
            {
                message: "Game will be having maintenance.",
                title: "Status update",
                type: NotificationType.Info
            }
        ]
    });

    const contextProps: INotificationContextProps = {
        addNotification: (notification: INotificationItem) => {
            const newNotifications = [...state.notifications];
            newNotifications.push(notification);

            const newState: INotificationContextProviderState = {
                notifications: newNotifications
            };

            setState(prevState => {
                return { ...prevState, ...newState };
            })
        },
        removeNotification: (notification) => {
            const notifications = [...state.notifications];
            const index = notifications.indexOf(notification);

            if (index > -1) {
                notifications.splice(index, 1);

                const newState: INotificationContextProviderState = {
                    notifications: notifications
                };

                setState(prevState => {
                    return { ...prevState, ...newState };
                })
            }
        },
        notifications: state.notifications
    };

    return (
        <NotificationContext.Provider
            value={contextProps}
            {...props}
        />
    );
}

function useNotificationContext() {
    return useContext(NotificationContext);
}

export { NotificationContextProvider, useNotificationContext };
