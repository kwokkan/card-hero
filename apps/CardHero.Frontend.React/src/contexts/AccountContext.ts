import React from "react";
import { IUserModel } from "../clients/clients";

interface IAccountContextProps {
    user?: IUserModel;
    setUser: (user: IUserModel) => void;
}

export const AccountContext = React.createContext<IAccountContextProps>({ setUser: (_user: IUserModel) => { } });