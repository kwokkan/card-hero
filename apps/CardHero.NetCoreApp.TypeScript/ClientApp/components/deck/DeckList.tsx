import React from "react";
import { Link } from "react-router-dom";
import { DeckModel } from "../../clients/clients";
import Icon from "../../styles/index";

interface IDeckListProps {
    decks: DeckModel[];
}

export default function DeckList(props: IDeckListProps) {
    return (
        <div className="row">
            <table className="table table-striped">
                <thead className="thead-inverse">
                    <tr>
                        <th>Name</th>
                    </tr>
                </thead>

                <tbody>
                    {props.decks.map(d =>
                        <tr key={d.id}>
                            <th scope="row">
                                <Icon icon="star" className={'deck-favourite' + (d.isFavourited ? ' enabled' : '')} data-deck-id={d.id} />
                                {' '}
                                <Link to={'/' + d.id}>{d.name}</Link>
                            </th>
                        </tr>
                    )}
                </tbody>
            </table>
        </div>
    );
}