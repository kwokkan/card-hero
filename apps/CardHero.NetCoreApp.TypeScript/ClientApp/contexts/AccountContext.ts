import React from "react";
import { UserModel } from "../clients/clients";

interface IAccountContextProps {
    user?: UserModel;
    setUser: (user: UserModel) => void;
}

export const AccountContext = React.createContext<IAccountContextProps>({ setUser: (_user: UserModel) => { } });