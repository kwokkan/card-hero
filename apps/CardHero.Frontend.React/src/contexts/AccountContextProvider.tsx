import { ProviderProps, useContext, useState } from "react";
import { IUserModel } from "../clients/clients";
import { AccountContext, IAccountContextProps } from "./AccountContext";

export interface IAccountContextProviderProps {
    user?: IUserModel;
}

interface IAccountContextProviderState {
    user?: IUserModel;
}

function AccountContextProvider(props: ProviderProps<IAccountContextProviderProps>): JSX.Element {
    const [state, setState] = useState<IAccountContextProviderState>({
        user: props.value.user
    });

    const contextProps: IAccountContextProps = {
        setUser: (user) => {
            setState({
                user: user
            });
        },
        user: state.user
    };

    return (
        <AccountContext.Provider
            value={contextProps}
            children={props.children}
        />
    );
}

function useAccountContext() {
    return useContext(AccountContext);
}

function useAccountContextConsumer() {
    return AccountContext.Consumer;
}

export { AccountContextProvider, useAccountContext, useAccountContextConsumer };
