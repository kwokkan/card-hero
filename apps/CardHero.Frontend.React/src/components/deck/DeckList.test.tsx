import React from "react";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { IDeckModel } from "../../clients/clients";
import { AccountContext, IAccountContextProps } from "../../contexts/AccountContext";
import { DeckList } from "./DeckList";

test("<DeckList /> renders correctly", () => {
    const decks: IDeckModel[] = [
        {
            id: 1,
            isFavourited: false,
            name: "Test deck"
        },
        {
            id: 2,
            isFavourited: true,
            name: "Test deck 2"
        }
    ];
    const contextValue: IAccountContextProps = {
        user: {
            id: 999
        },
        setUser: () => { }
    };
    const tree = renderer
        .create(
            <AccountContext.Provider value={contextValue}>
                <MemoryRouter>
                    <DeckList decks={decks} routePrefix="/jest/" />
                </MemoryRouter>
            </AccountContext.Provider>
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
