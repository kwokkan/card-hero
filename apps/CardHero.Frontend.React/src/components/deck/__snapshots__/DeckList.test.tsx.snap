// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeckList /> renders correctly 1`] = `
<div
  className="row"
>
  <table
    className="table table-striped"
  >
    <thead
      className="thead-inverse"
    >
      <tr>
        <th>
          Name
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 fa-fw deck-favourite"
            data-deck-id={1}
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           
          <a
            href="/jest/1"
            onClick={[Function]}
          >
            Test deck
          </a>
        </th>
      </tr>
      <tr>
        <th
          scope="row"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star fa-w-18 fa-fw deck-favourite enabled"
            data-deck-id={2}
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           
          <a
            href="/jest/2"
            onClick={[Function]}
          >
            Test deck 2
          </a>
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;
